### ISSUE TITLE
Regression: Now() = Today() should succeed

### ISSUE THREAD

**Issue by MikeStall** [microsoft/Power-Fx#1379]
This was recently regressed: 

```
> Now() = Today()
One or more errors occurred. (Errors: Error 6-7: Incompatible types for comparison. These types can't be compared: DateTime, Date.)
```

`Now() = Today()` should succeed.  

Likely #1372  (which was a re-revert from #1281)

**Comment by gregli-msft** [comment]
Note that not all comparison operations are broken:

```
> DateTime(2000,1,1,1,1,1) > Date(2023,3,27)
false

> DateTime(2000,1,1,1,1,1) < Date(2023,3,27)
true

> DateTime(2000,1,1,1,1,1) = Date(2023,3,27)
One or more errors occurred. (Errors: Error 25-26: Incompatible types for comparison. These types can't be compared: DateTime, Date.)

> DateTime(2000,1,1,1,1,1) <> Date(2023,3,27)
One or more errors occurred. (Errors: Error 25-27: Incompatible types for comparison. These types can't be compared: DateTime, Date.)

> If(1<0, DateTime(2000,1,1,1,1,1), Date(2023,3,27))
DateTime(2023,3,27,0,0,0,0)
```

**Comment by CarlosFigueiraMSFT** [comment]
Fixed by #1381
